data/moves.js

```diff
@@ -14757,10 +14757,11 @@ let BattleMovedex = {
 		beforeTurnCallback(pokemon) {
 			pokemon.addVolatile('shelltrap');
 		},
-		beforeMoveCallback(pokemon) {
+		onTry(pokemon) {
```

Comment:
Oh yeah! This was exactly what I tried when I attempted to fix it and it failed spectacularly because `Try` runs against each target, so when Shell Trap succeeds or fails it does so multiple times.

Reply:
Just sayin', #5216 fixes that.

Reply:
@MacChaeger - fixes it so that 0962c84 would actually work and not have the problems @Marty-D describes?

Reply:
Well, events like `Try` only run against one target in my version.

Reply:
Gotcha. I tried to repro the '`Try` runs against each target' bug Marty described in 034fbac / c06192a, but the test still seems to pass? It seems like Shell Trap is only succeeding once, at least according to the damage its doing and the amount of PP its losing?

Reply:
Hmm, maybe I misremembered the success part since it doesn't happen twice currently, but `cant` is indeed sent twice when it fails.

Reply:
I've modified the test to check the log for the double `cant`. Just to verify: https://gist.github.com/scheibo/c9a2a78708420b53f93678a8370fd2a6 is the log after the second turn, where `cant` shows up twice - does it look invalid?

I'm going to leave the move with `beforeMoveCallback` with a TODO for the time being (and the test `skip`ed) - once @MacChaeger lands his spread move fix we can switch to using `onTry` and reenable the test.